src/OpenLoco/src/Map/MapGenerator/PngTerrainGenerator.cpp

```diff
@@ -12,40 +12,38 @@ using namespace OpenLoco::Diagnostics;
 
 namespace OpenLoco::World::MapGenerator
 {
-    void PngTerrainGenerator::generate(const S5::Options& options, const fs::path& path, HeightMapRange heightMap)
+    void PngTerrainGenerator::generate(const S5::Options& options, const fs::path& path, HeightMap& heightMap)
     {
         if (!fs::is_regular_file(path))
         {
-            Logging::error("Can't find terrain file ({})", path);
+            Logging::error("Can't find heightmap file ({})", path);
             return;
         }
 
         auto pngImage = PngOps::loadPng(path.string());
         if (pngImage == nullptr)
         {
-            Logging::error("Can't load terrain file ({})", path);
+            Logging::error("Can't load heightmap file ({})", path);
             return;
         }
 
         const int maxHeightmapLevels = 64 - options.minLandHeight;
         const float scalingFactor = maxHeightmapLevels / 255.f;
 
-        for (int32_t y = 0; y < World::kMapRows; y++)
+        std::fill_n(heightMap.data(), heightMap.size(), options.minLandHeight);
+
+        auto width = std::min((int)World::kMapColumns, pngImage->width);
+        auto height = std::min((int)World::kMapRows, pngImage->height);
```

Comment:
Please avoid C-style casts:
```suggestion
        auto width = std::min<int>(World::kMapColumns, pngImage->width);
        auto height = std::min<int>(World::kMapRows, pngImage->height);
```

Reply:
fixed